GeoLib/OctTree-impl.h

```diff
@@ -0,0 +1,260 @@
+/**
+ * \date   2015-06-12
+ * \brief  Implementation of the OctTree class.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+
+namespace GeoLib {
+
+template <typename POINT>
+OctTree<POINT> OctTree<POINT>::createOctTree(POINT ll, POINT ur,
+	double eps, std::size_t max_items_per_node)
+{
+	OctTree<POINT>::_max_items_per_node = max_items_per_node;
+	OctTree<POINT>::_eps = eps;
+	// compute an axis aligned cube around the points ll and ur
+	const double dx(ur[0] - ll[0]);
```

Comment:
I do not see how the AABB can help at this place. Here a special AABB is computed - a cube. The input points `ll` and `ur` needs not to describe a cube.